cmake/cpu_extension.cmake

```diff
@@ -68,18 +69,28 @@ if (AVX512_FOUND AND NOT AVX512_DISABLED)
     else()
         message(WARNING "Disable AVX512-BF16 ISA support, no avx512_bf16 found in local CPU flags." " If cross-compilation is required, please set env VLLM_CPU_AVX512BF16=1.")
     endif()
+    
 elseif (AVX2_FOUND)
     list(APPEND CXX_COMPILE_FLAGS "-mavx2")
     message(WARNING "vLLM CPU backend using AVX2 ISA")
+    
 elseif (POWER9_FOUND OR POWER10_FOUND)
     message(STATUS "PowerPC detected")
     # Check for PowerPC VSX support
     list(APPEND CXX_COMPILE_FLAGS
         "-mvsx"
         "-mcpu=native"
         "-mtune=native")
+
+elseif (ASIMD_FOUND)
+    message(STATUS "ARMv8 architecture detected")
+    list(APPEND CXX_COMPILE_FLAGS
+        "-mcpu=native"
+        "-mtune=native"
```

Comment:
Native is OK for local development, however the binaries won't be portable at all.

I actually thought you were on the right track before this change. We really ought to explicitly specify a fairly minimal base ARM architecture (maybe ARMv8.4) and then explicitly the set of extensions that we need to build (BF16, and maybe FP16 and DotProd?)

Reply:
I see, I'll try to set it as armv8.4 as a base implementation. Then, depending on the flags it would run bf16 or not.

Reply:
Could you try using `"-march=armv8.4-a+bf16+dotprod+fp16"`?

Reply:
Just for your guys reference, on arm64 -mcpu= acts as both specifying the appropriate architecture and tuning and it's generally better to use that vs -march if you're building for a specific CPU, you can find more details from here if you are running any test on Graviton: https://github.com/aws/aws-graviton-getting-started/blob/main/c-c++.md

Reply:
> Could you try using `"-march=armv8.4-a+bf16+dotprod+fp16"`?

Yes, I tried it and it works. It even works with `"-march=armv8.2-a+bf16+dotprod+fp16"` making it compatible with Graviton2.

Reply:
@tlrmchlsmth I'll fix `"-march=armv8.2-a+bf16+dotprod+fp16"` for this PR, and make it backward compatible with mac chips if the issue is raised for that in future PR. What do you suggest?